File: Problem045-Jump Game II/Solution.java
/*
 * @SEU(master);@ZZULI(bachelor)
 * author:CuiXuan
 * email:873059043@qq.com
 * If you find any errors, please let me know.
 * If you have any better solution, please let me know.
 * 
 * Solution Using time 396ms
 */
public class Solution {
    public int jump(int[] nums) {
        if(nums.length <= 1)return 0;
        int[] jumps = new int[nums.length];
        jumps[0]=0;
        int pos=0;
        for(int i=0;i<nums.length;i++){
            if(i+nums[i] > pos){
                if(i+nums[i] >= nums.length-1)return jumps[i]+1;
                for(int j=pos+1;j<=i+nums[i];j++){
                    jumps[j] = jumps[i]+1;
                }
                pos = i+nums[i];
            }
        }
        return jumps[nums.length-1];
    }
}

File: Problem045-Jump Game II/Solution.py
"""
 @SEU(master);@ZZULI(bachelor)
 author:CuiXuan
 email:873059043@qq.com
 If you find any errors, please let me know.
 If you have any better solution, please let me know.

 Solution Using time 80ms
"""
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:return 0
        jumps = [0 for i in range(len(nums))]
        pos=0
        for i in range(len(nums)):
            if i+nums[i] > pos:
                if i+nums[i] >= len(nums)-1:return jumps[i]+1
                for j in range(pos+1,i+nums[i]+1):
                    jumps[j] = jumps[i]+1
                pos = i+nums[i]
        return jumps[len(nums)-1]